### ISSUE TITLE
Removing DataSet from LineData while Marker View is showing

### ISSUE THREAD

**Issue by atnegrete** [PhilJay/MPAndroidChart#3332]
I get a null pointer at 
`
int com.github.mikephil.charting.interfaces.datasets.IDataSet.getEntryIndex(com.github.mikephil.charting.data.Entry)`

I have tried setting the Marker to not showing on the LineChart before removing the DataSet, but I get the error once I show it again. I feel like there's probably a way around this, but I am not familiar how. Maybe I don't understand the flow of rendering completely, hope it's not a dumb issue!

Thanks!

**Comment by mhanifsp** [comment]
Use `yourChart.clear()` before removing DataSets

**Comment by atnegrete** [comment]
Yes, but I am not trying to remove  all the DataSets, just one. 

`chart.clear` would remove the entire LineData if I'm correct?

**Comment by mhanifsp** [comment]
Yes it would I think. And it removes the marker too. The workaround i did with this problem is defining DataSets in `ArrayList<ILineDataSet>` in my Class. Just use chart.clear(), remove the unneeded DataSet from the list and assign the list again to the chart.

You can read MultiLineChart example for ArrayList<ILineDataSets>

**Comment by mhanifsp** [comment]
By the way I just encountered with the problem a couple hours ago and searched through the source code. I found this workaround after reading the Chart.java source code. Please tell me if you found better way. Thanks ^^

**Comment by atnegrete** [comment]
That's exactly what I ended up doing actually! Thank you though, I will defeinetly let you know if I figure out something more straightforward.  Hopefully in the future there is an implementation for clearing a DataSet, or that `removeDataSet()` clears it automatically.
